### ISSUE TITLE
`image_mean` type problem

### ISSUE THREAD

**Issue by BassyKuo** [YangZhang4065/AdaptationSeg#12]
Hi, thanks for your great work! It's really impressive that you do the segmentation translation from the virtual world to the real world scene. But when I run your code, I have a little problem with your `image_mean` type. It seems that you loaded images in RGB (https://github.com/YangZhang4065/AdaptationSeg/blob/master/warp_data.py#L10L25) and subtracted image mean value `[103.939, 116.779, 123.68]`  from them (https://github.com/YangZhang4065/AdaptationSeg/blob/master/train_val_FCN_DA.py#L134). 

However, in the VGG network, they used image mean value `[103.939, 116.779, 123.68]` in BGR, not RGB (https://gist.github.com/ksimonyan/211839e770f7b538e2d8#description). I just wondered that why you do the operation of RGB-images with BGR-image_mean. (Maybe your SYNTHIA images are saved in BGR type before loading into the model?)
I would be very grateful if you could resolve my doubt. 
Thank you so much. :)

**Comment by YangZhang4065** [comment]
Hi @BassyKuo ,

Thanks for asking! This is actually a bug inherited from the previous VGG-FCN implementation, which has the same problem. We do load images in RGB using PIL. And indeed it would be better if we use the RGB mean `[123.68, 116.779, 103.939]` instead of BGR mean `[103.939, 116.779, 123.68]`. But 1, since we trained all of our proposed models/baselines on the `[103.939, 116.779, 123.68]` and got equivalent baseline performance reported by other paper 2, it won't influent our conclusion and relative performance, we left it untouched. However, you are more than welcome to train it with the reversed RGB mean and share your performance.

**Comment by BassyKuo** [comment]
@YangZhang4065 
Thanks for your reply! I run your `test_FCN_DA.py` by using RGB images and BGR images. Just like you said, it shows that the RGB one makes better performance.
I am going to re-train the SYNTHIA_FCN model with the BGR image mean. Could you kindly provide steps how do you train it? (Do I need to prepare a pre-train FCN model first, and finetune it with the SYNTHIA dataset?) I will share the new model result if I trained it successfully.
Thanks again.:)

**Comment by YangZhang4065** [comment]
@BassyKuo I did not use a pretrained  FCN because most pretrained FCN is trained using real dataset and that would unfairly boost our experiment. However, I did manually use standard ImageNet VGG19 to initialize the backbone weights and you can do it as well. Unfortunately, I am unable to provide the VGG weights now as I do not have access to the server so you will need to find an VGG19 weights to initialize it or you can try a random initialization.

After that, just remove label distribution loss, superpixel loss and cityscape loading (You do not need cityscape images during baseline training). And train the model with only the segmentation loss on SYNTHIA. I recommend you to use my training framework in this repo. The CityScape validation in this training framework would ensure you to save the baseline model which maximizes your performance on CityScape dataset.

**Comment by BassyKuo** [comment]
Thanks a lot. I will try it!
